pallets/anchors-v2/src/tests.rs

```diff
@@ -0,0 +1,350 @@
+		});
+	}
+
+	#[test]
+	fn unsigned_origin() {
+		new_test_ext().execute_with(|| {
+			let document_id = 123;
+			let document_version = 456;
+			let hash = H256::random();
+
+			assert_err!(
+				AnchorsV2::set_anchor(RawOrigin::None.into(), document_id, document_version, hash,),
+				BadOrigin
+			);
+		});
+	}
+
+	#[test]
+	fn anchor_present() {
+		new_test_ext().execute_with(|| {
+			let origin: u64 = 1;
+			let document_id = 123;
+			let document_version = 456;
+			let hash = H256::random();
+			let deposit = AnchorDeposit::<Runtime>::get();
+
+			let anchor = AnchorOf::<Runtime> {
+				account_id: origin,
+				document_id,
+				document_version,
+				hash,
+				deposit,
+			};
+
+			Anchors::<Runtime>::insert((document_id, document_version), origin, anchor);
+
+			assert_err!(
+				AnchorsV2::set_anchor(
+					RuntimeOrigin::signed(origin),
+					document_id,
+					document_version,
+					hash,
+				),
+				Error::<Runtime>::AnchorAlreadyExists
+			);
+		});
+	}
+
+	#[test]
+	fn personal_anchor_present() {
+		new_test_ext().execute_with(|| {
+			let origin: u64 = 1;
+			let document_id = 123;
+			let document_version = 456;
+			let hash = H256::random();
+			let deposit = AnchorDeposit::<Runtime>::get();
+
+			let anchor = AnchorOf::<Runtime> {
+				account_id: origin,
+				document_id,
+				document_version,
+				hash,
+				deposit,
+			};
+
+			PersonalAnchors::<Runtime>::insert((origin, document_id, document_version), anchor);
+
+			assert_err!(
+				AnchorsV2::set_anchor(
+					RuntimeOrigin::signed(origin),
+					document_id,
+					document_version,
+					hash,
+				),
+				Error::<Runtime>::PersonalAnchorAlreadyExists
+			);
+		});
+	}
+
+	#[test]
+	fn insufficient_balance() {
+		new_test_ext().execute_with(|| {
+			let origin: u64 = 1;
+			let document_id = 123;
+			let document_version = 456;
+			let hash = H256::random();
+
+			assert_err!(
+				AnchorsV2::set_anchor(
+					RuntimeOrigin::signed(origin),
+					document_id,
+					document_version,
+					hash,
+				),
+				InsufficientBalance::<Runtime>
+			);
+		});
+	}
+}
+
+mod remove_anchor {
+	use super::*;
+
+	#[test]
+	fn success() {
+		new_test_ext().execute_with(|| {
+			let origin: u64 = 1;
+			let document_id = 123;
+			let document_version = 456;
+			let hash = H256::random();
+			let deposit = AnchorDeposit::<Runtime>::get();
+
+			let anchor = AnchorOf::<Runtime> {
+				account_id: origin,
+				document_id,
+				document_version,
+				hash,
+				deposit,
+			};
+
+			Anchors::<Runtime>::insert((document_id, document_version), origin, anchor.clone());
+			PersonalAnchors::<Runtime>::insert((origin, document_id, document_version), anchor);
+
+			let account_info = AccountInfo::<_, AccountData<Balance>> {
+				nonce: 0,
+				consumers: 0,
+				providers: 0,
+				sufficients: 0,
+				data: AccountData::<Balance> {
+					free: 0,
+					reserved: deposit,
+					frozen: 0,
+					flags: Default::default(),
+				},
+			};
+
+			Account::<Runtime>::insert(origin, account_info);
```

Comment:
Out of curiosity, why not `Balance::reserve(origin, deposit)` here?

Reply:
Copy/pasta from keystore tests ^^

Gonna change this and other similar case.